zone/pets.cpp

```diff
@@ -405,6 +405,19 @@ void Mob::MakePoweredPet(uint16 spell_id, const char* pettype, int16 petpower,
 
 	entity_list.AddNPC(npc, true, true);
 	SetPetID(npc->GetID());
+	// save updated clients petinfo
+	this->CastToClient()->Save(1);
```

Comment:
This isn't safe.

Reply:
I should wrap inside 
if (this->IsClient()) { 

per say:
https://github.com/EQEmu/Server/blob/master/zone/groups.cpp#L324